testsuite/integration-arquillian/tests/base/src/main/java/org/keycloak/testsuite/arquillian/containers/KeycloakQuarkusServerDeployableContainer.java

```diff
@@ -47,20 +48,20 @@
 import org.jboss.shrinkwrap.api.exporter.ZipExporter;
 import org.jboss.shrinkwrap.descriptor.api.Descriptor;
 import org.keycloak.testsuite.arquillian.SuiteContext;
+import org.keycloak.testsuite.model.StoreProvider;
 
 /**
  * @author mhajas
  */
 public class KeycloakQuarkusServerDeployableContainer implements DeployableContainer<KeycloakQuarkusConfiguration> {
 
     private static final int DEFAULT_SHUTDOWN_TIMEOUT_SECONDS = 10;
-    private static final String AUTH_SERVER_QUARKUS_MAP_STORAGE_PROFILE = "auth.server.quarkus.mapStorage.profile.config";
 
     private static final Logger log = Logger.getLogger(KeycloakQuarkusServerDeployableContainer.class);
 
     private KeycloakQuarkusConfiguration configuration;
     private Process container;
-    private static AtomicBoolean restart = new AtomicBoolean();
+    private static AtomicBoolean firstExecution = new AtomicBoolean(true);
```

Comment:
I'm not 100% sure if the best here is to change the logic around this flag.

The main idea around that flag is to only run re-augmentation (implicit `build`) when the server is restarting. This eventually happens during test execution due to changes to features, deploying providers, or changing anything else that requires a  `build` prior to starting the server. Also, note that the logic around this flag also makes sure the same build options are set as when running the Ant plugin to pre-configure the distribution (auth-server-quarkus).

Instead, I would change the Ant script (`configure.xml` in auth-server-quarkus module) to pre-configure the distribution with the choosen database vendor and then make sure the Arquillian container includes the database vendor when doing restarts.

Reply:
@pedroigor Thank you very much for your comment!

I've only renamed the flag as I think it makes more sense in the current context of determining whether the build phase should be executed, or not. To be more specific, it emphasizes the fact that's necessary to execute the build phase only for the first time when some particular DB or store is required; not for the default H2 DB. 
However, if you think it makes more sense to keep the var name `restart`, I can revert it (it's just "negated" logic for better readability).

Regarding the ant script, I wouldn't say it's a better approach. The old approach is suitable only for the default H2 DB, where it's not necessary to execute the build phase after the first initialization of the Quarkus server; even for this new approach, there's no build phase, when the H2 should be used. Each store (Legacy, CHM, Map-JPA, HotRod) requires different arguments to run, in order to properly execute tests with it, and IMHO, setting the configuration in the codebase is better than having it set up by some Maven properties. 

Moreover, from the dev perspective, as it's kinda important to execute tests with ease, it'd bring some additional effort and complications, when the DB vendor should be set up already in the ant script. It'd mean when a developer would try to execute some tests with different DB vendors (f.e PostgreSQL and then MariaDB), he/her would have to always build the whole Quarkus auth server (/servers/auth-server/...). ATM, a developer can just execute the base module with the specified DB profile - f.e `mvn clean install -f /testsuite/integration-arquillian/tests/base/pom.xml -Dtest=UsersTest -Pauth-server-quarkus,db-postgres`.

Reply:
Reverted `restart` logic.

Reply:
Yeah, that is true. By pre-configuring the distribution with a database (default to H2/no-vendor) we add an additional step where the `auth-server-quarkus` module should be built prior to executing tests.

However, my argument is all about reducing test execution time. I understand that for H2 tests it should not make a difference but when running DB tests the time will increase significantly.

For now, we can focus on enabling the DB tests for Quarkus. And then evaluate later improvements to execution time. Not hard on fixing this now.

Reply:
@pedroigor I understand your concerns, but IMO, the proper execution of tests with various DBs is more important ATM than the execution time. And as you've mentioned, we can improve the execution time of those tests in some follow-up tasks :) Thank you!